pkg/module/ModuleService.go

```diff
@@ -291,15 +326,54 @@ func (impl ModuleServiceImpl) HandleModuleAction(userId int32, moduleName string
 	module.Status = ModuleStatusInstalling
 	module.Version = moduleActionRequest.Version
 	module.UpdatedOn = time.Now()
+	tx, err := impl.moduleRepository.GetConnection().Begin()
+	if err != nil {
+		impl.logger.Errorw("error in  opening an transaction", "err", err)
+		return nil, err
+	}
+	defer tx.Rollback()
+	res := strings.Split(moduleName, ".")
+	toolName := strings.ToUpper(res[1])
+	flagForEnablingState := false
+	if len(moduleActionRequest.ModuleType) == 0 {
+		flagForEnablingState = true
+	} else {
+		// Finding the Module by type and status, if no module exists of current type marking current module as active and enabled by default.
+		err = impl.moduleRepository.FindByModuleTypeAndStatus(moduleActionRequest.ModuleType, ModuleStatusInstalled)
+		if err != nil {
+			if err == pg.ErrNoRows {
+				var toolversion string
+				if moduleName == ModuleNameSecurityClair {
+					toolversion = CLAIR_V4
+				} else if moduleName == ModuleNameSecurityTrivy {
+					toolversion = TRIVY_V1
+				}
+				err2 := impl.scanToolMetaDataRepository.MarkToolAsActive(toolName, toolversion, tx)
+				if err2 != nil {
+					impl.logger.Errorw("error in marking tool as active ", "err", err)
```

Comment:
add metadata in logs

Reply:
why are we logging err here instead of err2 ??